EEBOS/SCROWL
# [SC-320] - Implement Export process in scrowl

### Short summary

- For now we are using the fork from [this repo](https://github.com/andre-dietrich/simple-scorm-packager)
   - We can create our own fork in a later step after using it a bit more and see a need to extend 
- Created an example button in the renderer that will invoke the export process in the main process
-

---

### Test steps


Before starting please have the LMS going locally: https://www.notion.so/LMS-Local-Setup-ab83bca28bd04d61947a4a54e55b5c4b

---

NOTE: You cannot export while in docker so you will need to test this on host machine (permission conflict between the export process which needs root and electron which cannot start if the container has root access )

Once the app is up locally on host machine:

1. Create a folder called `res` in `course-authoring/src/scormBuild`
1. In that folder create a file called `start.html`
2. In `start.html` file add the following:
```
<!DOCTYPE html>
<html>
<head>
    <meta charset='utf-8'>
    <meta http-equiv='X-UA-Compatible' content='IE=edge'>
    <title>Page Title</title>
    <meta name='viewport' content='width=device-width, initial-scale=1'>
</head>
<body>
    <h1>Helo world!</h1>
</body>
</html>
```
3. Click the `export` button in the electron app and you should see a new zip folder in `src/scormPackages`
4. Insure that the zip file works by uploading it into a local LMS instance:
   1. Please follow these Instructions: https://www.notion.so/Test-SCORM-Package-041b79520b5042459d1cddc7208255a5
